src/main/java/io/github/thebusybiscuit/slimefun4/implementation/tasks/RadioactivityTask.java

```diff
@@ -0,0 +1,100 @@
+import org.bukkit.potion.PotionEffect;
+import org.bukkit.potion.PotionEffectType;
+
+import javax.annotation.Nonnull;
+
+/**
+ * The {@link RadioactivityTask} handles radioactivity for
+ * {@link Radioactive} items.
+ *
+ * @author Semisol
+ *
+ */
+public class RadioactivityTask implements Runnable {
+
+    private static final RadiationSymptom[] SYMPTOMS = RadiationSymptom.values();
+    private static StatusEffect RADIATION_EFFECT;
+
+    {
+        assert SlimefunPlugin.instance() != null;
+        RADIATION_EFFECT = new StatusEffect(new NamespacedKey(SlimefunPlugin.instance(), "radiation"));
+    }
+
+    public static void removePlayer(@Nonnull Player p) {
+        RADIATION_EFFECT.clear(p);
+    }
+
+    @Override
+    public void run() {
+        for (Player p : Bukkit.getOnlinePlayers()) {
+            if (!p.isValid() || p.isDead()) {
+                continue;
+            }
+
+            PlayerProfile.get(p, profile -> handleRadiation(p, profile));
+        }
+    }
+
+    private void handleRadiation(@Nonnull Player p, @Nonnull PlayerProfile profile) {
+        if (p.getGameMode() == GameMode.CREATIVE || p.getGameMode() == GameMode.SPECTATOR) {
+            return;
+        }
+
+        int exposureTotal = 0;
+
+        if (!profile.hasFullProtectionAgainst(ProtectionType.RADIATION)) {
+            for (ItemStack item : p.getInventory()) {
+                if (item == null || item.getType().isAir()) {
+                    continue;
+                }
+                SlimefunItem sfItem = SlimefunItem.getByItem(item);
+                if (sfItem instanceof RadioactiveItem) {
+                    exposureTotal += item.getAmount() * ((RadioactiveItem) sfItem).getRadioactivity().getExposureModifier();
+                }
+            }
+        }
+        
+        int exposureLevelBefore = RADIATION_EFFECT.getLevel(p).orElse(0);
+        if (exposureTotal > 0) {
+            if (exposureLevelBefore == 0) {
+                SlimefunPlugin.getLocalization().sendMessage(p, "messages.radiation");
+            }
+            RADIATION_EFFECT.addPermanent(p, Math.min(exposureLevelBefore + exposureTotal, 100));
+        } else if (exposureLevelBefore > 0) {
+            RADIATION_EFFECT.addPermanent(p, RADIATION_EFFECT.getLevel(p).orElse(1) - 1);
+        }
+        
+        int exposureLevelAfter = RADIATION_EFFECT.getLevel(p).orElse(0);
+        for (RadiationSymptom symptom : SYMPTOMS) {
+            if (symptom.shouldApply(exposureLevelAfter)) {
+                symptom.apply(p);
+            }
+        }
+        
+        if (exposureLevelAfter > 0 || exposureLevelBefore > 0) {
+            String msg = SlimefunPlugin.getLocalization().getMessage(p, "actionbar.radiation")
+                    .replace("%level%", "" + exposureLevelAfter);
+            p.spigot().sendMessage(ChatMessageType.ACTION_BAR,
+                    new ComponentBuilder().append(ChatColors.color(msg)).create()
+            );
```

Comment:
Pretty sure we already have an actionbar sending method in the localization class

Reply:
Does not have a way to replace a dynamic param from what I see.

Reply:
hmmm, alright then, something to note for the future.